src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -38,36 +38,46 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo> {
     ),
     this.content$.pipe(this.childrenNavData())
   );
-  state$ = combineLatest([ this.children$, this.content$ ]).pipe(
-    mergeScan((prevState: FetchState<NavTreeData>, [ children, content ]) => {
-
+  state$ = combineLatest([
+    this.children$,
+    this.content$.pipe(switchMap(content => this.currentContent.contentHasChanged$.pipe(
+      map(() => true),
+      startWith(false),
+      map(hasChanged => ({ content, hasChanged }))
+    ))),
+  ]).pipe(
+    mergeScan((prevState: FetchState<NavTreeData>, [ children, { content, hasChanged }]) => {
       // CASE 1: the current-content does not match the type of this nav tree (so `content` has been mapped to `undefined`)
       if (!content) {
+        if (hasChanged) return this.fetchDefaultNav().pipe(filter(state => state.isReady), startWith(fetchingState(prevState.data)));
         // CASE 1A: the menu has already an element displayed -> just deselect what is selected if there was a selection
         if (prevState.isReady) return of(readyState(prevState.data.withNoSelection()));
         // CASE 1B: the menu has nothing displayed yet -> load item root
-        else return this.fetchDefaultNav();
+        return this.fetchDefaultNav();
 
       } else {
       // CASE 2: the content type matches the type of this nav tree
         const route = content.route;
 
-        if (prevState.isReady && prevState.data.hasElement(route)) {
-          // CASE 2A : the content is among the displayed elements -> either select it if at root or shift the tree "to the left" otherwise
+        if (hasChanged) {
+          // CASE 2A: reload content after content change
+          return this.fetchNewNav(content).pipe(
+            mapStateData(data => this.mapNewNavData(data, children, content)),
+            filter(state => state.isReady),
+            startWith(fetchingState(prevState.data)),
+          );
```

Comment:
Instead of : 
1. calling `fetchNewNav` which gets the appropriate nav data and maps it to a fetch state, so add loading state when data is not ready 
2. map the data to merge with current knowledge
3. keep only the ready state
4. add a custom fetching state which emits immediately (but what's the use if there is already the same data being displayed in "ready" state)

Why not:
1. calling a `getNewNav` which gets the appropriate nav data
2. map the data to merge with current knowledge
3. wrap the data in a ready state which will just emit the new state when it will be ready
And keep a `fetchNewNav`  which just call `getNewNav` and may it to a fetch state

Reply:
Fixed

Reply:
I do not see it fixed.

Reply:
Was missing in my "why not": error handling. 
Maybe allowing to use "mapToFetchState" without initial loading may be useful.